text/0580-destroyables.md

```diff
@@ -164,45 +166,42 @@ destructor from the destroyable.
 - Calling `unregisterDestructor` on a destroyed object should throw an error.
 - Calling `unregisterDestructor` with a destructor that is not associated with
   the object should throw an error.
-- Calling `unregisterDestructor` on a non-destroyable should throw an error.
 
 #### `destroy`
 
-`destroy` manually initiates the destruction of a destroyable. This allows users
-to manually manage the lifecycles of destroyables that are shorter lived than
-their parents (for instance, destroyables that are defined an a service, which
-lives as long as an application does), cleaning them up as needed.
+`destroy` initiates the destruction of an destroyable object. It runs all
+associated destructors, and then destroys all children recursively.
 
-Calling `destroy` multiple times on the same object is safe. It will not throw
-an error, and will not take any further action.
+Destruction via `destroy()` follows these steps:
 
-- Calling `destroy` on a non-destroyable should throw an error.
+1. Mark the destroyable such that `isDestroying(destroyable)` returns `true`
```

Comment:
Does it also mark each of the children?

Reply:
It does while calling `destroy()` on the children, since it recurses. It does not mark them as destroying before running its own destructors.